management/asset.yaml

```diff
@@ -0,0 +1,178 @@
+              schema:
+                $ref: './schemas/asset.yaml#/schemas/AssetList'
+  '/v1/asset/{assetId}':
+    parameters:
+      - name: assetId
+        in: path
+        description: Id of the Asset
+        required: true
+        schema:
+          type: integer
+          format: int32
+    get:
+      tags:
+        - assets
+      description: Get an existing Asset
+      operationId: getAsset
+      security:
+        - ApiKeyAuth: []
+      responses:
+        200:
+          description: The existing Asset
+          content:
+            application/json:
+              schema:
+                $ref: './schemas/asset.yaml#/schemas/AssetDetails'
+  '/v1/advertiser/{advertiserId}/asset':
+    parameters:
+      - name: advertiserId
+        in: path
+        description: Id of the Advertiser
+        required: true
+        schema:
+          type: integer
+          format: int32
+    get:
+      tags:
+        - assets
+      description: List Assets for Advertiser
+      operationId: listAdvertiserAssets
+      security:
+        - ApiKeyAuth: []
+      parameters:
+        - name: pageSize
+          in: query
+          description: The size of the page to be returned
+          required: false
+          schema:
+            type: integer
+            format: int32
+            nullable: true
+        - name: page
+          in: query
+          description: The page number to be returned
+          required: false
+          schema:
+            type: integer
+            format: int32
+            nullable: true
+      responses:
+        200:
+          description: Paged list of Assets for Advertiser
+          content:
+            application/json:
+              schema:
+                $ref: './schemas/asset.yaml#/schemas/AssetList'
+    post:
+      tags:
+        - assets
+      description: Create Asset
+      operationId: createAsset
+      security:
+        - ApiKeyAuth: []
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: './schemas/asset.yaml#/schemas/AssetInput'
+      responses:
+        200:
+          description: Asset Created
+          content:
+            application/json:
+              schema:
+                $ref: './schemas/asset.yaml#/schemas/Asset'
+  '/v1/asset/{assetId}/delete':
+    parameters:
+      - name: assetId
+        in: path
+        description: Id of the Asset
+        required: true
+        schema:
+          type: integer
+          format: int32
+    post:
+      tags:
+        - assets
+      description: Delete an existing Asset
+      operationId: deleteAsset
+      security:
+        - ApiKeyAuth: []
+      responses:
+        200:
+          description: Successfully Deleted
+  '/v1/asset/{assetId}/upload':
+    parameters:
+      - name: assetId
+        in: path
+        description: Id of the Asset
+        required: true
+        schema:
+          type: integer
+          format: int32
+    post:
```

Comment:
This also has some required query parameters, you can see from this example path generated by the api below. Required query params are:

* timestamp: integer + required
* signature: string + required

```
"[HOST]/v1/asset/[assetId]/upload?timestamp=1234567&signature=aabbccddeeffgg"
```

Reply:
Thank you so much for capturing this one and walking me through the process to properly test all these.